lib/perl/Genome/Model/Command/Admin/RemoveDiskAllocationsFromTestdb.pm

```diff
@@ -0,0 +1,308 @@
+package Genome::Model::Command::Admin::RemoveDiskAllocationsFromTestdb;
+
+use strict;
+use warnings;
+
+use Genome;
+use DBI;
+use TestDbServer::CmdLine qw(search_databases get_template_by_id get_database_by_id);
+use Genome::Config;
+use Try::Tiny;
+use Sub::Install qw();
+use Sub::Name qw();
+
+use constant KB_IN_ONE_GB => 1024 * 1024;
+use constant KB_IN_ONE_MB => 1024;
+
+class Genome::Model::Command::Admin::RemoveDiskAllocationsFromTestdb {
+    is => 'Command::V2',
+    doc => 'Remove disk allocations that were created while running under a test database',
+    has => [
+        database_name => {
+            is => 'Text',
+            default_value => _get_default_database_name(),
+            doc => 'test database name, derived from the ds_gmschema_server config value',
+        },
+        database_server => {
+            is => 'Text',
+            default_value => _get_default_database_server(),
+            doc => 'database server name, derived from the ds_gmschema_server config value',
+        
+        },
+        database_port => {
+            is => 'Text',
+            default_value => _get_default_database_port(),
+            doc => 'database listening port, derived from the ds_gmschema_server config value',
+        },
+        template_name => {
+            is => 'Text',
+            default_value => _get_default_template_name(),
+            doc => 'template database name, resolved from the database host in the ds_gmschema_server config value',
+        },
+    ],
+};
+
+sub _get_default_database_name {
+    my $conn = _parse_database_connection_info_from_env_var();
+    return $conn->{dbname};
+}
+
+sub _get_default_database_server {
+    my $conn = _parse_database_connection_info_from_env_var();
+    return $conn->{host};
+}
+
+sub _get_default_database_port {
+    my $conn = _parse_database_connection_info_from_env_var();
+    return( $conn->{port} // 5432 );
+}
+
+sub _get_default_template_name {
+    my $test_db_name = __PACKAGE__->_get_default_database_name();
+    return __PACKAGE__->get_template_name_for_database_name($test_db_name);
+}
+
+sub _parse_database_connection_info_from_env_var {
+    my %connection;
+    foreach my $key ( qw( dbname host port ) ) {
+        no warnings 'uninitialized';
+        my $server = Genome::Config::get('ds_gmschema_server');
+        ($connection{$key}) = $server =~ m/$key=(.*?)(?:;|$)/;
+    }
+    return \%connection;
+}
+
+sub execute {
+    my $self = shift;
+
+    unless ($self->is_running_in_test_env) {
+        die $self->error_message('Must be run within a test environment created with genome-test-env');
+    }
```

Comment:
I think these will running under genome-env, not genome-test-env, because we want to be committing.  Or maybe I am unclear on the consequences of genome-test-env vs. genome-env?